care/emr/api/viewsets/medication_request.py

```diff
@@ -41,5 +46,28 @@ def get_queryset(self):
             .select_related("patient", "encounter", "created_by", "updated_by")
         )
 
+    def authorize_create(self, instance):
+        super().authorize_create(instance)
+        if instance.requester:
+            encounter = get_object_or_404(Encounter, external_id=instance.encounter)
+            requester = get_object_or_404(User, external_id=instance.requester)
+            if not AuthorizationController.call(
+                "can_update_encounter_obj", requester, encounter
+            ):
+                raise PermissionDenied(
+                    "Requester does not have permission to update encounter"
+                )
+
+    def authorize_update(self, request_obj, model_instance):
```

Comment:
Lets not allow the requester to be updated, let them delete and recreate

Reply:
done